src/compiler/clang.rs

```diff
@@ -35,6 +35,10 @@ use crate::errors::*;
 pub struct Clang {
     /// true iff this is clang++.
     pub clangplusplus: bool,
+    /// true iff this is Apple's clang(++).
+    pub is_appleclang: bool,
+    /// String from __VERSION__ macro.
+    pub version: Option<String>,
```

Comment:
Should we consider adding `version` always?

Reply:
Moved `version` from CCompiler to every `CCompilerImpl` struct in https://github.com/mozilla/sccache/pull/1162/commits/3f0880ae577786b6fb6bfc98276db1281e38f046. The alternative would have been to pass the version argument through a lot of functions, but semantically the value fits better to `CCompilerImpl` anyway, next to `is_clang`, `is_applyclang`, etc.

Reply:
Apologize for my inaccuracy, I meant: Can we drop the `Option<_>` wrapping since `__VERSION__` is always evaluated?

Reply:
`detect_c_compiler` has a case to detect when the `__VERSION__` macro has not been replaced, in which case no version information would be available and the current implementation passes `None` to `CCompiler.version`. I could use an empty string in this case (and the tests that passed `None` so far), if that's preferable?

Reply:
Fair enough, I'd say leave as is - thanks for the elaborate clarification!